solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -243,25 +253,85 @@ contract WalletRegistry is
         address notifier
     );
 
+    // Custom Errors
+
+    // Authorization Errors
+
+    /// @notice Raised when caller is not the staking contract or allowlist contract.
+    error CallerNotStakingContract();
```

Comment:
now with allow list might be better to change name to `CallerNotAuthorizationSource` or similar